src/http/http_jwt.h

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "crypto/hash.h"
+#include "http_consts.h"
+#include "http_parser.h"
+#include "tls/base64.h"
+#include "tls/key_pair.h"
+
+#define FMT_HEADER_ONLY
+#include <fmt/format.h>
+#include <optional>
+#include <string>
+
+namespace http
+{
+  enum class JwtCryptoAlgorithm
+  {
+    RS256
+  };
+  DECLARE_JSON_ENUM(JwtCryptoAlgorithm, {{JwtCryptoAlgorithm::RS256, "RS256"}});
+
+  struct JwtHeader
+  {
+    JwtCryptoAlgorithm alg;
+    std::string kid;
+  };
+  DECLARE_JSON_TYPE(JwtHeader)
+  DECLARE_JSON_REQUIRED_FIELDS(JwtHeader, alg, kid)
+
+  class JwtVerifier
+  {
+  public:
+    struct Token
+    {
+      nlohmann::json header;
+      JwtHeader header_typed;
```

Comment:
Do we need to keep both of these around? If we can get away with it, I think we should store just the JSON `header` in here - the verifier can extract whatever fields it needs when verifying, but these don't need to be retained from parsing?

Reply:
It roughly goes: parse JWT, get keyid from token, lookup cert in kv matching keyid, validate token using cert. So, you need a typed variant to get the key id out. Of course I could move the following into `frontend.h` but that's suboptimal since I wanted all the parsing logic in http_jwt.h:
```
      JwtHeader header_typed;
      try
      {
        header_typed = header.get<JwtHeader>();
      }
      catch (const nlohmann::json::exception& e)
      {
        error_reason =
          fmt::format("JWT header does not follow schema: {}", e.what());
        return std::nullopt;
      }
```